### ISSUE TITLE
jax.scipy.linalg.solve(assume_a="pos") with batched `a` triggers its own cho_solve batched-1D deprecation FutureWarning

### ISSUE THREAD

**Issue by s-sasaki-earthsea-wizard** [jax-ml/jax#39986]
## Description

Since jax 0.10.0, calling `jax.scipy.linalg.solve(a, b, assume_a="pos")` with a batched `a` emits the `cho_solve` batched-1D deprecation `FutureWarning`, even though user code never calls `cho_solve` and is using the documented `solve` API in a non-deprecated way. The warning is triggered by `_solve`'s own internal use of `cho_solve`.

```
FutureWarning: jax.scipy.linalg.cho_solve: batched 1D solves with b.ndim > 1 are deprecated, and in the future will be treated as a batched 2D solve. Use cho_solve(c_and_lower, b[..., None]).squeeze(-1) to avoid this warning.
```

## Minimal reproduction

```python
import warnings
import jax
import jax.numpy as jnp
from jax.scipy.linalg import solve

B = jax.random.normal(jax.random.PRNGKey(0), (3, 4, 6, 6))
G = B @ jnp.swapaxes(B, -1, -2) + 6 * jnp.broadcast_to(jnp.eye(6), (3, 4, 6, 6))  # SPD stack
Id = jnp.broadcast_to(jnp.eye(6), (3, 4, 6, 6))

with warnings.catch_warnings(record=True) as w:
    warnings.simplefilter("always")
    out = solve(G, Id, assume_a="pos")          # batched a, matrix b
print(sum(issubclass(x.category, FutureWarning) for x in w))   # 1  (expected: 0)
```

The batched-vector case (`solve(G, v, assume_a="pos")` with `v` of shape `(3, 4, 6)`) goes through the same internal call and also warns on a fresh trace. (If you run it in the same process *after* the matrix example, the warning may not appear again — the internal `cho_solve` is `@jit`-cached for that shape — so reproduce the two cases in separate processes or call `jax.clear_caches()` in between.)

Results are numerically correct in all cases (checked against `jnp.linalg.inv` / residuals); only the warning is spurious. The warning also fires (once per trace) when the call happens inside `jit`.

## Mechanism

In `jax/_src/scipy/linalg.py`, `_solve` wraps the factorization in `custom_linear_solve` with

```python
solve=lambda _, x: cho_solve(factors, x)
```

- For matrix `b` (`a.ndim == b.ndim`), `_solve` vmaps `custom_solve` over the trailing column axis, so inside the vmap the per-example rhs has `x.ndim == factors.ndim - 1`.
- For batched-vector `b` (`a.ndim == b.ndim + 1`), the non-vmap branch passes `b` with `b.ndim == factors.ndim - 1` directly.

Both paths satisfy `_cho_solve`'s deprecation condition (`c.ndim == b.ndim + 1 and c.shape[-1] == b.shape[-1]`). That deprecation was added 2026-03-23 in the disambiguation work around #36083 — and `_solve`'s own internal dispatch is exactly the ambiguous pattern it targets, so jax warns on itself rather than on user code.

## Suggested fix

Use the form the warning message itself recommends inside `_solve`:

```python
solve=lambda _, x: cho_solve(factors, x[..., None]).squeeze(-1)
```

This routes every internal solve through the unambiguous matrix branch. It behaves identically for all three shape cases (`a` unbatched + vector `b`, `a` batched + vector `b`, and the per-column vmapped case), and is compatible with the post-deprecation semantics.

## Downstream impact

Projects running pytest with `filterwarnings = ["error"]` fail their suites as soon as their environment picks up jax >= 0.10.0. For example, isce-framework/dolphin currently fails 44 tests on an unmodified checkout because `dolphin/phase_link/crlb.py` calls `solve(..., assume_a="pos")` on batched matrices — tracked there as isce-framework/dolphin#739.

## System info

- Present in jax 0.10.0 through `main` (as of 2026-08-14); the last release without the deprecation warning is 0.9.2
- Reproduced on jax/jaxlib 0.10.1 (conda-forge, CUDA container) and 0.10.2 (pip, CPU)
- Python 3.13, Linux x86_64

---

Disclosure: AI tools assisted with research, implementation, and drafting. I verified the technical claims and take responsibility for the proposal.

**Comment by jakevdp** [comment]
Thanks for the report! I'll take a look at the issue.
